packages/base/card-api.gts

```diff
@@ -2073,25 +2073,31 @@ export async function recompute(
     );
     do {
       for (let fieldName of [...pendingFields]) {
-        let value = await getIfReady(
-          model,
-          fieldName as keyof T,
-          undefined,
-          opts
-        );
-        if (!isNotReadyValue(value) && !isStaleValue(value)) {
-          pendingFields.delete(fieldName);
-          if (recomputePromises.get(card) !== recomputePromise) {
-            return;
-          }
-          if (Array.isArray(value)) {
-            for (let item of value) {
-              if (item && isCard(item) && !stack.includes(item)) {
-                await _loadModel(item, [item, ...stack]);
+        let field = getField(
+          Reflect.getPrototypeOf(model)!.constructor as typeof Card,
+          fieldName as string
+        ) as Field;
+        if (field.computeVia) {
+          let value = await getIfReady(
+            model,
+            fieldName as keyof T,
+            undefined,
+            opts
+          );
+          if (!isNotReadyValue(value) && !isStaleValue(value)) {
+            pendingFields.delete(fieldName);
+            if (recomputePromises.get(card) !== recomputePromise) {
+              return;
+            }
+            if (Array.isArray(value)) {
+              for (let item of value) {
+                if (item && isCard(item) && !stack.includes(item)) {
+                  await _loadModel(item, [item, ...stack]);
+                }
               }
+            } else if (isCard(value) && !stack.includes(value)) {
+              await _loadModel(value, [value, ...stack]);
```

Comment:
I don't think this is the right place to make this change, I think you need to go deeper down to where the recompute is actually called. specifically inside of the `getIfReady`. moreover `getIfReady` needs to set the deserialized bucket data for _all_ fields. right now the `getIfReady` has a default function for the `compute()` that just returns the field value. if we decide to not run the actual recompute for the fields are are not computeds, we'll still need to populate the deserialized data bucket for these field values.

Reply:
Updated https://github.com/cardstack/boxel/pull/314/commits/9d28ac0bddda33d8759c93b0f03d0814b8803044

Reply:
Updated https://github.com/cardstack/boxel/pull/314/commits/9d28ac0bddda33d8759c93b0f03d0814b8803044